File: 2018/java essentials/chapters/09_interfaces/src/be/pxl/h9/oefeningen/oef1/Point.java
package be.pxl.h9.oefeningen.oef1;

public class Point {
    public static int MAXIMUM_SIZE = 99;
    private int x;
    private int y;

    public Point(int x, int y) {
        this.x = Math.min(Math.abs(x), MAXIMUM_SIZE);
        this.y = Math.min(Math.abs(y), MAXIMUM_SIZE);
    }

    public void setX(int x) {
        this.x = Math.min(Math.abs(x), MAXIMUM_SIZE);
    }

    public void setY(int y) {
        this.y = Math.min(Math.abs(y), MAXIMUM_SIZE);
    }

    public int getX() {
        return x;
    }

    public int getY() {
        return y;
    }

    public double getDistanceOf(Point point) {
        return Math.sqrt(Math.pow(this.x - point.x, 2) + Math.pow(this.y - point.y, 2));
    }

    public String toString() {
        return String.format("x:%d y:%d", this.x, this.y);
    }
}

File: 2019/web advanced/phpstorm/chapters/04_oop/voorbeeldenLes/src/drawing/Point.php
<?php

namespace drawing;

class Point
{
    public const MIN = 0;
    public const MAX = 100;
    private static int $count = 0;
    private int $x;
    private int $y;

    public function __construct(int $x,int $y)
    {
        $this->validateXY($x);
        $this->validateXY($y);
        $this->x = $x;
        $this->y = $y;
        self::$count++;
    }

    public function __destruct()
    {
        self::$count--;
    }

    private function validateXY(int $value): void
    {
        if ($value < self::MIN || $value >= self::MAX) {
            throw new \Exception("$value is not between $this::MIN and $this::MAX");
        }
    }

    public function getX(): int
    {
        return $this->x;
    }
    
    public function getY(): int
    {
        return $this->y;
    }

    public function setX(int $x): void
    {
        $this->validateXY($x);
        $this->x = $x;
    }

    public function setY(int $y): void
    {
        $this->validateXY($y);
        $this->y = $y;
    }

    public static function getCount(): int
    {
        return self::$count;
    }
    
    public static function makeOrigin(): Point
    {
        return new self(0,0);
    }

    public function __toString(): string
    {
        return "($this->x, $this->y)";
    }
}